### PR TITLE
Maximum Sum Subarray - C

### PR DESCRIPTION
**Pull request by SangeetaMishra**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#555

### CODE REVIEW

File: C/README.md [hunk 1]
```diff
@@ -23,7 +23,8 @@
 - [Randomized selection algorithm](search/randomized_selection_algorithm.c)
 - [Sum of Triplet](cp/sum_of_triplet.c)
 - [Calculate Double Factorial](cp/doublefactorial.c)
-- [Longest substring without repeating characters](cp/Longest substring without repeating chars.cpp)
+- [Longest substring without repeating characters](cp/Longest%20substring%20without%20repeating%20chars.cpp)
```
Comment (iamrajiv) [inline C/README.md]: @SangeetaMishra I have the updated the name `Longest substring without repeating characters.cpp` program to `Longest_substring_without_repeating_chars.cpp` so update this to `- [Longest substring without repeating characters](cp/Longest_substring_without_repeating_chars.cpp)`

File: C/cp/max_sum_subarray.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    return max(max(presum, postsum), presum + postsum);
+
+}
+
+int max_sum_subarray(int arr[], int lo, int hi) {
+    if(lo == hi) {
+        return arr[lo];
+    }
+    int mid = (lo + hi) / 2;
```
Comment (iamrajiv) [inline C/cp/max_sum_subarray.c]: @SangeetaMishra Use `mid = lo + ((hi - lo) / 2)` to avoid integer overflow condition.

**Review by iamrajiv — CHANGES_REQUESTED** [review]

**Review by iamrajiv — APPROVED** [review]

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Review by Kajol-Kumari — APPROVED** [review]
LGTM :+1:

**Pull request merged**